tokio/tests/async_send_sync.rs

```diff
@@ -265,6 +267,17 @@ async_assert_fn!(tokio::sync::watch::Sender<u8>::closed(_): Send & Sync);
 async_assert_fn!(tokio::sync::watch::Sender<Cell<u8>>::closed(_): !Send & !Sync);
 async_assert_fn!(tokio::sync::watch::Sender<Rc<u8>>::closed(_): !Send & !Sync);
 
+async_assert_fn!(tokio::sync::OnceCell<u8>::get_or_init(
+    _, fn() -> Pin<Box<dyn Future<Output = u8> + Send>>): Send & !Sync);
+async_assert_fn!(tokio::sync::OnceCell<Cell<u8>>::get_or_init(
+    _, fn() -> Pin<Box<dyn Future<Output = Cell<u8>>>>): !Send & !Sync);
+async_assert_fn!(tokio::sync::Lazy<u8>::get(_): Send & !Sync);
+async_assert_fn!(tokio::sync::Lazy<Cell<u8>>::get(_): !Send & !Sync);
+assert_value!(tokio::sync::OnceCell<u8>: Send & Sync);
+assert_value!(tokio::sync::OnceCell<Rc<u8>>: !Send & !Sync);
```

Comment:
We need to go through all of `u8`, `Cell<u8>` and `Rc<u8>` here to check all three cases:

 1. `Send & Sync`
 2. `Send & !Sync`
 3. `!Send & !Sync`

In fact, you are missing the most interesting case!